src/viam/services/mlmodel/utils.py

```diff
@@ -37,7 +39,14 @@ def make_ndarray(flat_data, dtype, shape):
         """Takes flat data (protobuf RepeatedScalarFieldContainer | bytes) to output an ndarray
         of appropriate dtype and shape"""
         make_array = np.frombuffer if dtype == np.int8 or dtype == np.uint8 else np.array
-        return make_array(flat_data, dtype).reshape(shape)
+        # as per proto, int16 and uint16 are stored as uint32. As of numpy v2, this creates
+        # some strange interactions with negative values for int16 specifically. creating
```

Comment:
Mild preference to capitalize the first word of each sentence (both lines 42 and 43)

Reply:
Done!